web-server/v0.4/src/utils/parse.js

```diff
@@ -197,7 +190,7 @@ export const parseIterationData = results => {
               var child_column = columns[parentColumnIndex].children[childColumnIndex];
               if (!containsIteration(child_column, columnTitle)) {
                 if (child_column['children'] == undefined) {
-                  child_column['children'] = [{ title: columnTitle, dataIndex: columnTitle }];
+                  child_column['children'] = [{ title: columnTitle, dataIndex: _columnPrefix }];
```

Comment:
Why does `_columnPrefix` work here?  Seems like that is just a local variable used to avoid duplicate code up at line 177.

Reply:
We need `_columnPrefix` here as opposed to `columnTitle` as there were console warnings for potential duplicate keys given a dataIndex of `columnTitle`. By using the whole column prefix, we ensure that table cells have unique dataIndex values by referencing the iteration primary metric and network type they belong to.

Reply:
Not sure why we have an underscore in `_columnPrefix` but I'll modify that so we maintain the camel case conventions we have throughout here.

Reply:
I believe I wrote that code to indicate that it was a local temporary variable not intended to be used elsewhere.  It was done to avoid repeating code multiple times (DRY principle).

So why that would help still is a mystery.

Reply:
It could be possible that a recent update to antd's library modified the behavior of tables and required unique dataIndex values which is why we were not receiving those warnings at that time. Here's the warning for reference: 

`Warning: Encountered two children with the same key, "client_hostname:100.100.100.138-server_hostname:100.100.100.140-server_port:20010". Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.`

You'll see here we only use `columnTitle` which only contains hostname information whereas the `columnPrefix` contains both the information in `columnTitle` and the iteration type and network. This causes potential issues for iteration types and networks that are indexed with similar hostname data.

Reply:
Okay, so we should rename `columnPrefix` since it is not just a prefix, but a unique identifier.

Reply:
I've renamed the reference here to `columnKey` which references the `columnHost` information we collect. I've also included a cleaner solution for maintaining immutability of result iterations data when filtering and improved the naming of references for better oversight into what is being copied and filtered vs. the original source reference.